### PR TITLE
Add Algeria holidays

### LINKED ISSUE #1145: Add Algeria holidays
Useful links:
  - https://en.wikipedia.org/wiki/Public_holidays_in_Algeria
  - https://www.timeanddate.com/holidays/algeria/

### PR DESCRIPTION
**Pull request by abh31000**
## Proposed change

Added Algeria holidays
Closes #1145 
Stuck at 82% coverage. Can anyone tell me what I'm missing here?

## Type of change

- [x] New country/market holidays support (thank you!)
- [ ] Supported country/market holidays update (calendar discrepancy fix, localization)
- [ ] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency upgrade (version update)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] This PR is filed against `beta` branch of the repository
- [x] This PR doesn't contain any merge conflicts and has clean commit history
- [x] The code style looks good: `make pre-commit`
- [x] All tests pass locally: `make test`, `make tox` (we strongly encourage adding tests to your code)
- [x] The related [documentation][docs] has been added/updated (check off the box for free if no updates is required)

[contributing-guidelines]: https://github.com/dr-prodigy/python-holidays/blob/beta/CONTRIBUTING.rst
[docs]: https://github.com/dr-prodigy/python-holidays/tree/beta/docs/source

### CODE REVIEW

**Comment by KJhellico** [comment]
@abh31000, thank you for your contribution!

Seems your code based on `master` branch. You need to merge latest `beta` and add new line to `holidays/registry.py` for Algeria.

File: holidays/countries/algeria.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        # In January 2018, Algeria declared Yennayer a national holiday
+        if year >= 2018:
+            self._add_holiday("Amazigh New Year", JAN, 12)
+
+        # Labour Day
+        self._add_labor_day("Labour Day")
+
+        # Independance Day
+        if year >= 1962:
+            self._add_holiday("Independance Day", JUL, 5)
```
Comment (KJhellico) [inline holidays/countries/algeria.py:35]:
```suggestion
        # Independence Day
        if year >= 1962:
            self._add_holiday("Independence Day", JUL, 5)
```

File: holidays/countries/algeria.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+        elif year > 2023:
+            # Eid al-Fitr - Feast Festive
+            self._add_eid_al_fitr_day("Eid al-Fitr")
+            self._add_eid_al_fitr_day_two("Eid al-Fitr Holiday")
+            self._add_eid_al_fitr_day_three("Eid al-Fitr Holiday")
+            # Eid al-Adha - Scarfice Festive
+            self._add_eid_al_adha_day("Eid al-Adha")
+            self._add_eid_al_adha_day_two("Eid al-Adha Holiday")
+            self._add_eid_al_adha_day_three("Eid al-Adha Holiday")
```
Comment (KJhellico) [inline holidays/countries/algeria.py:77]:
```suggestion
        # Eid al-Fitr - Feast Festive
        self._add_eid_al_fitr_day("Eid al-Fitr")
        self._add_eid_al_fitr_day_two("Eid al-Fitr Holiday")
        if year >= 2024:
            self._add_eid_al_fitr_day_three("Eid al-Fitr Holiday")

        # Eid al-Adha - Scarfice Festive
        self._add_eid_al_adha_day("Eid al-Adha")
        self._add_eid_al_adha_day_two("Eid al-Adha Holiday")
        if year >= 2023:
            self._add_eid_al_adha_day_three("Eid al-Adha Holiday")
```

File: holidays/countries/algeria.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+        # Labour Day
+        self._add_labor_day("Labour Day")
+
+        # Independance Day
+        if year >= 1962:
+            self._add_holiday("Independance Day", JUL, 5)
+
+        # Revolution Day
+        if year > 1962:
```
Comment (KJhellico) [inline holidays/countries/algeria.py:38]:
```suggestion
        if year >= 1963:
```

File: holidays/countries/algeria.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+        # Labour Day
+        self._add_labor_day("Labour Day")
+
+        # Independence Day
+        if year >= 1962:
+            self._add_holiday("Independence Day", JUL, 5)
+
+        # Revolution Day
+        if year > 1962:
```
Comment (KJhellico) [inline holidays/countries/algeria.py:38]:
```suggestion
        if year >= 1963:
```

File: tests/countries/test_algeria.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    def test_2022(self):
+        self.assertHoliday("2022-01-01")
+        self.assertHoliday("2022-01-12")
+        self.assertHoliday("2022-05-01")
+        self.assertHoliday("2022-05-02")
+        self.assertHoliday("2022-05-03")
+        self.assertHoliday("2022-07-05")
+        self.assertHoliday("2022-07-09")
+        self.assertHoliday("2022-07-10")
+        self.assertHoliday("2022-11-01")
```
Comment (arkid15r) [inline tests/countries/test_algeria.py:22]: The TestCase:assertHoliday was designed to accept multiple date-like strings:
```suggestion
        self.assertHoliday(
            "2022-01-01",
            "2022-01-12",
            "2022-05-01",
            "2022-05-02",
            "2022-05-03",
            "2022-07-05",
            "2022-07-09",
            "2022-07-10",
            "2022-11-01",
        )
```

Plese group them where possible.

File: holidays/countries/algeria.py [hunk 6]
```diff
@@ -0,0 +1,70 @@
+from holidays.constants import JAN, JUL, NOV
+from holidays.holiday_base import HolidayBase
+from holidays.holiday_groups import IslamicHolidays, InternationalHolidays
+
+
+class Algeria(HolidayBase, IslamicHolidays, InternationalHolidays):
```
Comment (arkid15r) [inline holidays/countries/algeria.py:6]:
```suggestion
class Algeria(HolidayBase, InternationalHolidays, IslamicHolidays):
```

**Review by arkid15r — COMMENTED** [review]
Hi @abh31000,
thank you so much for implementing the Algeria holidays!

We're excited to see a new contributor! Also thanks a lot for addressing all the comments in such a timely manner.

I left a couple of minor comments below. Otherwise your code looks good!

P.S. as for the coverage issue you had with your first commit of this PR -- you can find out the  uncovered lines by running `make coverage` (which is an alias for `pytest --cov=. --cov-config=pyproject.toml --cov-report term-missing --dist loadscope --no-cov-on-fail --numprocesses auto`)

**Comment by abh31000** [comment]
> Hi @abh31000, thank you so much for implementing the Algeria holidays!
> 
> We're excited to see a new contributor! Also thanks a lot for addressing all the comments in such a timely manner.
> 
> I left a couple of minor comments below. Otherwise your code looks good!
> 
> P.S. as for the coverage issue you had with your first commit of this PR -- you can find out the uncovered lines by running `make coverage` (which is an alias for `pytest --cov=. --cov-config=pyproject.toml --cov-report term-missing --dist loadscope --no-cov-on-fail --numprocesses auto`)

You're welcome!
And now with this commit the test coverage is 100%.

**Review by arkid15r — COMMENTED** [review]
🇩🇿 LGTM

**Pull request merged**